lighthouse-core/scripts/timings.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+// Example:
+//     node lighthouse-core/scripts/timings.js --name my-collection --collect -n 3 --lh-flags='--only-audits=unminified-javascript' --urls https://www.example.com https://www.nyt.com
+//     node lighthouse-core/scripts/timings.js --name my-collection --summarize --measure-filter 'loadPage|connect'
+
+const fs = require('fs');
+const {execSync} = require('child_process');
+const yargs = require('yargs');
+
+const LH_ROOT = `${__dirname}/../..`;
+const ROOT_OUTPUT_DIR = `${LH_ROOT}/timings-data`;
+
+const argv = yargs
+  .help('help')
+  .describe({
+    'name': 'Unique identifier, makes the folder for storing LHRs. Not a path',
+    // --collect
+    'collect': 'Saves LHRs to disk',
+    'lh-flags': 'Lighthouse flags',
+    'urls': 'Urls to run',
+    'n': 'Number of times to run',
+    // --summarize
+    'summarize': 'Prints statistics report',
+    'measure-filter': 'Regex filter of measures to report. Optional',
+    'output': 'table, json',
+  })
+  .string('measure-filter')
+  .default('output', 'table')
+  .array('urls')
+  .string('lh-flags')
+  .default('lh-flags', '')
+  // Why is the printing for examples so awful?
+  // eslint-disable max-len
+  // .example('node lighthouse-core/scripts/timings.js --name my-collection --collect -n 3 --lh-flags=\'--only-audits=unminified-javascript\' --urls https://www.example.com', 'Collect')
+  // .example('node lighthouse-core/scripts/timings.js --name my-collection --summarize --measure-filter \'loadPage|connect\'', 'Summarize')
+  // eslint-enable max-len
+  .wrap(yargs.terminalWidth())
+  .argv;
+
+const outputDir = `${ROOT_OUTPUT_DIR}/${argv.name}`;
+
+if (argv.collect) {
```

Comment:
might be nice to have a `main()`, `collect()` and `summarize()` to separate this core logic from the helpers :)